lib/sender.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const net = require('net')
+const tls = require('tls')
+const { Writable } = require('stream')
+
+
+/**
+ * Event sender.
+ * Can be used as a stream, with the usual functions:
+ * sender.write(), sender.end(), etc.
+ * Implements backpressure.
+ */
+class Sender extends Writable {
+
+  /**
+   * Create the client.
+   * @param {Object} options for sender, including:
+   *  - host: the host to send events, default localhost.
+   *  - port: the destination port (mandatory).
+   *  - objectMode: whether to allow sending objects.
+   * @param {function(*)} callback (Optional) Function to invoke when the
+   * connection has been established.
+   */
+  constructor(options, callback) {
+    super(options)
+    this._options = options
+    const libnet = options.cert ? tls : net
+    this._socket = libnet.connect(options, callback)
+    this._socket.on('error', error => this.emit('error', error))
+    //this._socket.unref()
+  }
+
+  /**
+   * Send an event to the ingestion endpoint.
+   *
+   * @param {string|Buffer|Object} event message to send.
+   * If an object is passed it will be converted to JSON before sending.
+   * @param {function(*)} callback Optional function to invoke with (error).
+   * @return true if the stream accepts more events.
+   */
+  send(event, callback) {
+    const converted = this._convert(event)
+    return this._socket.write(converted, null, error => {
```

Comment:
Not sure about the nomenclature you are using here. An `event` sounds to me like a `syslog event`, containing the timestamp, host, tag and rest of stuff that is added in `this._transforn`. I would change `event` by `content` and then `transformed` (line 51) by `event`. Just a proposal, I am not sure about what's the proper syslog nomenclature.

Reply:
I have changed all `event` variables in commit 6a24fd5. You are right, it is best not to use it to avoid misunderstandings.